src/azul/plugins/metadata/hca/transform.py

```diff
@@ -773,6 +775,21 @@ def _is_dcp2_matrix(self, file: api.File) -> bool:
         else:
             return False
 
+    def file_source(self, submitter_id: str) -> Optional[str]:
+        """
+        Return the human-readable version of the name that was used to generate
+        the submitter UUID.
+        """
+        try:
+            return self.contributor_submitter_ids[submitter_id][1]
```

Comment:
The fixed tuple indices are only OK in very close proximity to the tuple definition. This was borderline for me. It's just hard to read and understand what `[1]` is, if it weren't for the docstring.